docarray/base_document/document.py

```diff
@@ -213,3 +217,43 @@ def _group_fields(doc: 'BaseDocument') -> _FieldGroups:
             elif dict1 is not None and dict2 is not None:
                 dict1.update(dict2)
                 setattr(self, field, dict1)
+
+    @classmethod
+    def smart_parse_obj(cls: Type[T], obj: Dict[str, Any]) -> T:
+        """
+        parse the given object, validate the data and return a new initialized objet
+        the difference from `parse_obj` is that it will smartly resolve conflict and
+        undefined key.
+
+        Here we should describe the algo and give an example:
+        :param obj, a dict of field and type
+        :return: a Document initialized from the dict data
+        """
```

Comment:
so obj is being parsed to the schema of cls?

Reply:
yes